src/coremotion.cs

```diff
@@ -859,4 +865,117 @@ interface CMRecordedPressureData {
 		NSDate StartDate { get; }
 	}
 
+	[iOS (16,0), Watch (9,0), NoMac, NoTV, MacCatalyst (16,0)]
+	[Native]
+	public enum CMWaterSubmersionDepthState : long
+	{
+		Unknown = 0,
+		NotSubmerged = 100,
+		SubmergedShallow = 200,
+		SubmergedDeep = 300,
+		ApproachingMaxDepth = 400,
+		PastMaxDepth = 500,
+		SensorDepthError = 600,
+	}
+
+	[iOS (16,0), Watch (9,0), NoMac, NoTV, MacCatalyst (16,0)]
+	[Native]
+	public enum CMWaterSubmersionState : long
+	{
+		Unknown = 0,
+		NotSubmerged,
+		Submerged,
+	}
+
+	interface ICMWaterSubmersionManagerDelegate {}
+	[iOS (16,0), Watch (9,0), NoMac, NoTV, MacCatalyst (16,0)]
+#if NET
+	[Protocol, Model]
+#else
+	[Protocol, Model (AutoGeneratedName = true)]
+#endif
+	[BaseType (typeof (NSObject))]
+	interface CMWaterSubmersionManagerDelegate
+	{
+		[Abstract]
+		[Export ("manager:didUpdateEvent:")]
+		void DidUpdateEvent (CMWaterSubmersionManager manager, CMWaterSubmersionEvent @event);
+
+		[Abstract]
+		[Export ("manager:didUpdateMeasurement:")]
+		void DidUpdateMeasurement (CMWaterSubmersionManager manager, CMWaterSubmersionMeasurement measurement);
+
+		[Abstract]
+		[Export ("manager:didUpdateTemperature:")]
+		void DidUpdateTemperature (CMWaterSubmersionManager manager, CMWaterTemperature measurement);
+
+		[Abstract]
+		[Export ("manager:errorOccurred:")]
+		void DidErrorOccurred (CMWaterSubmersionManager manager, NSError error);
```

Comment:
```suggestion
		void ErrorOccurred (CMWaterSubmersionManager manager, NSError error);
```

Reply:
@rolfbjarne This one was originally "ErrorOccurred" but I attempted to Verbify it. Perhaps "DidErrorOccur~ed~" would have been better but I'm curious why to use "ErrorOccurred" here

Reply:
Mostly because that's what the selector says. I also thought about `DidErrorOccur`, but that's pretty much the same as `ErrorOccurred`, and in that case I prefer whatever the selector says to make things easier.